frontend/packages/console-shared/src/components/markdown-highlight-extension/MarkdownHighlightExtension.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { extension } from 'showdown';
+import { Spotlight } from '../spotlight';
+
+extension('quickstart', () => {
+  return [
+    {
+      type: 'lang',
+      regex: /\[(\w+)]{{(highlight)\s([\w-]+)}}/g,
+      replace: (text: string, linkLabel: string, linkType: string, linkId: string): string => {
+        return `<button class="pf-c-button pf-m-inline pf-m-link" data-highlight="${linkId}">${linkLabel}</button>`;
+      },
+    },
+  ];
+});
+type MarkdownHighlightExtensionProps = {
+  docContext: HTMLDocument;
+};
+const MarkdownHighlightExtension: React.FC<MarkdownHighlightExtensionProps> = ({ docContext }) => {
+  const [selector, setSelector] = React.useState<string>(null);
+  React.useEffect(() => {
+    const elements = docContext.querySelectorAll('[data-highlight]');
+    function startHighlight(e) {
+      const highlightId = e.target.getAttribute('data-highlight');
+      if (!highlightId) {
+        return;
+      }
+      setSelector(null);
+      setTimeout(() => {
```

Comment:
Capture the handle and `clearTimeout` on effect dispose.

Reply:
done